### ISSUE TITLE
Loss is NAN

### ISSUE THREAD

**Issue by iko0167** [hank-ai/darknet#104]
When i try to train yolov4-crowdhuman dataset with yolov4 loss and avg_loss is always nan, but when i try to train same dataset with yolov4-tiny or yolov4-tiny-3l or even yolov7-tiny its ok.

In addition when i try to train same dataset with same system (Same CUDA and CUDNN version, same laptop) with AlexeyAB repo and yolov4 its working fine.

My System:

NVIDIA GeForce RTX 3070 Mobile 
12th Gen Intel i7-12700H 
CUDA Version 12.4 with drivers
CUDNN 8.9.7

**Comment by stephanecharette** [comment]
So far I cannot replicate.  I've tried both yolov4-tiny at 416x416 and yolov4 at 608x608.  I'm currently in the middle of training the YOLOv4 one.

You are using the up-to-date .cfg files, right?  And you're using the training command as described in the FAQ to train?

Are you also using DarkMark to setup the project?  I'm using DarkMark v1.10.16.

**Comment by iko0167** [comment]
I am using yolov4-custom.cfg under cfg folder on this repository. As training command I'm using the command as you mentioned in documentation of this repository whic is:
darknet detector train -map -dont_show example.data yolov4-example.cfg

The exact dataset im using is below:

https://github.com/jkjung-avt/yolov4_crowdhuman

**Comment by stephanecharette** [comment]
Yes, that is the dataset I downloaded as well.  It is working for me.  The difference I can see from the command you give above is I use DarkMark to manage the project, generate the files, calculate the YOLO anchors, and start training.

**Comment by iko0167** [comment]
Got it. Now I use DarkMark to prepare data for training and i realize that when i choose yolov4-custom.cfg, this nan issue is still there but when i use yolov4.cfg instead of yolov4-custom.cfg it semms to working well. I think its about the yolov4-custom.cfg

**Comment by stephanecharette** [comment]
If you diff the two config files, you'll see the learning rate is the big difference between the two.

**Comment by iko0167** [comment]
Ok i have some updates to inform you. If i select yolov4_custom.cfg i cannot train with any learning rate value. But when i switched to yolov4.cfg i can train with any learning rate (Increase,Decrease) without errors. I think its independent with learning rate because even if i increase the learning rate to 0.261 on yolov4.cfg i can train without with errors

**Comment by jpfleischer** [comment]
i am also sometimes getting this issue. i cannot reliably replicate this collapsing of the neural network, but I usually just try again, decreasing the network size. and it tends to stop collapsing and instead converges on a high mAP as intended.

**Comment by jpfleischer** [comment]
loss=NaN
https://github.com/pjreddie/darknet/issues/716
https://github.com/AlexeyAB/darknet/issues/588#issuecomment-379333746
https://github.com/pjreddie/darknet/issues/1461
https://github.com/pjreddie/darknet/issues/1560
https://github.com/pjreddie/darknet/issues/1233
https://github.com/pjreddie/darknet/issues/1831
https://github.com/pjreddie/darknet/issues/567
https://stackoverflow.com/questions/60158549/getting-nan-during-darknet-training-what-am-i-doing-wrong

**Comment by jpfleischer** [comment]
https://github.com/hank-ai/darknet/commit/52bfd6f266712057bb3eba569030ddb45a605f6e

this has made great headway into solving this

**Comment by jpfleischer** [comment]
https://github.com/AlexeyAB/darknet?tab=readme-ov-file#how-to-train-with-multi-gpu

> 
> ## How to train with multi-GPU
> 
> 1. Train it first on 1 GPU for like 1000 iterations: `darknet.exe detector train cfg/coco.data cfg/yolov4.cfg yolov4.conv.137`
> 
> 2. Then stop and by using partially-trained model `/backup/yolov4_1000.weights` run training with multigpu (up to 4 GPUs): `darknet.exe detector train cfg/coco.data cfg/yolov4.cfg /backup/yolov4_1000.weights -gpus 0,1,2,3`
> 
> If you get a Nan, then for some datasets better to decrease learning rate, for 4 GPUs set `learning_rate = 0,00065` (i.e. learning_rate = 0.00261 / GPUs). In this case also increase 4x times `burn_in =` in your cfg-file. I.e. use `burn_in = 4000` instead of `1000`.
> 
> https://groups.google.com/d/msg/darknet/NbJqonJBTSY/Te5PfIpuCAAJ

I am not sure whether we do this already but its just a note that I found from AlexeyAB

**Comment by stephanecharette** [comment]
> https://github.com/AlexeyAB/darknet?tab=readme-ov-file#how-to-train-with-multi-gpu
> 
> > ## How to train with multi-GPU
> > 
> > 1. Train it first on 1 GPU for like 1000 iterations: `darknet.exe detector train cfg/coco.data cfg/yolov4.cfg yolov4.conv.137`
> > 2. Then stop and by using partially-trained model `/backup/yolov4_1000.weights` run training with multigpu (up to 4 GPUs): `darknet.exe detector train cfg/coco.data cfg/yolov4.cfg /backup/yolov4_1000.weights -gpus 0,1,2,3`

Note this is no longer necessary when using the Hank.ai repo.  Darknet will automatically run on 1 GPU until burn-in has been reached, and then training will automatically re-start with all GPUs listed in the `-gpus #,#,#...` option.

**Comment by iko0167** [comment]
A little update

This line on yolov4_custom.cfg is causing that issue;

stopbackward=800

Its not about the learning rate on my situation
